components/cam/src/physics/cam/micro_p3.F90

```diff
@@ -2797,20 +2811,19 @@ subroutine rain_self_collection(rho,qr_incld,nr_incld,    &
       ! note there should be a factor of 6^(1/3), but we
       ! want to keep breakup threshold consistent so 'dum'
       ! is expressed in terms of lambda rather than mass-mean D
-
-      dum2 = (qr_incld/(pi*rhow*nr_incld))**thrd
+      dum2 = bfb_cbrt(qr_incld/(pi*rhow*nr_incld))
       if (dum2.lt.dum1) then
          dum = 1._rtype
       else if (dum2.ge.dum1) then
-         dum = 2._rtype-exp(2300._rtype*(dum2-dum1))
+         dum = 2._rtype-bfb_exp(2300._rtype*(dum2-dum1))
       endif
 
       if (iparam.eq.1) then
          nrslf = dum*kr*1.e-3_rtype*qr_incld*nr_incld*rho
       elseif (iparam.eq.2 .or. iparam.eq.3) then
          nrslf = dum*5.78_rtype*nr_incld*qr_incld*rho
       endif
-
+   
```

Comment:
Remove trailing whitespace. You may need to add trailing-ws highlighting to your editor.